base_layer/core/src/base_node/tari_pulse_service/mod.rs

```diff
@@ -116,21 +121,41 @@ impl TariPulseService {
         notify_passed_checkpoints: watch::Sender<bool>,
     ) {
         let mut interval = time::interval(self.config.check_interval);
-        let mut interval_failed = time::interval(Duration::from_millis(100));
+        let mut interval_failed = time::interval(Duration::from_secs(60));
+        let mut shutdown_signal = self.shutdown_signal.clone();
+
         loop {
-            let passed_checkpoints = match self.passed_checkpoints(&mut base_node_service).await {
-                Ok(passed) => passed,
-                Err(err) => {
-                    error!(target: LOG_TARGET, "Error checking if node passed checkpoints: {:?}", err);
-                    interval_failed.tick().await;
-                    continue;
-                },
-            };
+            tokio::select! {
+                _ = interval.tick() => {
+                    let passed_checkpoints = match self.passed_checkpoints(&mut base_node_service).await {
+                        Ok(passed) => passed,
+                        Err(err) => {
+                            error!(target: LOG_TARGET, "Failed to check if node has passed checkpoints: {:?}", err);
+                            let interval_in_secs = interval_failed.period().as_secs();
+                            if interval_in_secs > (60 * 30) {
+                                warn!(target: LOG_TARGET, "Reached maximum retry interval of 30 minutes. Exiting");
+                                break;
+                            }
+                            error!(target: LOG_TARGET, "Retrying in {} minutes", interval_in_secs/60);
+                            interval_failed.tick().await;
+                            interval_failed.tick().await;
+                            interval_failed = time::interval(Duration::from_secs(interval_in_secs * 2));
+                            continue;
+                        },
+                    };
```

Comment:
I think this could be simplified, something like
```suggestion
                _ = interval.tick() => {
                    let passed_checkpoints = match self.passed_checkpoints(&mut base_node_service).await {
                        Ok(passed) => {
                            interval_failed = time::interval(Duration::from_secs(0));
                            passed,
                        }
                        Err(err) => {
                            let interval_in_secs = interval_failed.period().as_secs();
                            if interval_in_secs > (60 * 30) {
                                warn!(target: LOG_TARGET, "Reached maximum retry interval of 30 minutes. Exiting");
                                break;
                            }
                            error!(target: LOG_TARGET, "Failed to check if node has passed checkpoints, will retry: {:?}", err);
                            interval_failed += interval;
                            continue;
                        },
                    };
```